lms/djangoapps/experiments/utils.py

```diff
@@ -305,6 +305,75 @@ def get_experiment_user_metadata_context(course, user):
     context['has_staff_access'] = has_staff_access
     context['forum_roles'] = forum_roles
     context['partition_groups'] = user_partitions
+
+    user_metadata = {
+        key: context.get(key)
+        for key in (
+            'username',
+            'user_id',
+            'course_id',
+            'enrollment_mode',
+            'upgrade_link',
+            'upgrade_price',
+            'audit_access_deadline',
+            'course_duration',
+            'pacing_type',
+            'has_staff_access',
+            'forum_roles',
+            'partition_groups',
+            # TODO: clean up as part of REVO-28 (START)
+            'has_non_audit_enrollments',
+            # TODO: clean up as part of REVO-28 (END)
+            # TODO: clean up as part of REVEM-199 (START)
+            'program_key_fields',
+            # TODO: clean up as part of REVEM-199 (END)
+        )
```

Comment:
(Unrelated to this ticket, but I'm confirming in #experimentation that these aren't being used in live experiments and safe to clean up)

Reply:
i think we can just leave them unless they're hard to implement, i think most or all of the user metadata fields are not currently being used, but if any of these came in handy in a past experiment then they might be helpful for future experiments

Reply:
Having extra fields we don't need could slow down performance, then again I don't have a good sense of what is likely to be used down the road and give value. Looking up those cleanup tickets, there were other spots to clean up as well, so I'll defer to those tickets for a more holistic decision.

Reply:
Should we take who is able to access the endpoint into consideration when figuring out which fields to expose? e.g. would anyone be able to  iterate over a list of usernames/courses to find out which account has staff access?

Reply:
Just as the user-metadata and made available to the DOM specifically for Optimizely experiments, this API is made for our Optimizely experiments to be able to retrieve the same set of info from the courseware MFE, so the user making this request will always be Optimizely. 
To confirm, are you asking about the case where a bad actor has a list of edx usernames and enrolled courses and is able to think up the endpoint uri, and make calls to it?

Reply:
Yes, that is the use case I'm thinking of.

Reply:
Update: the original permission restricts results to an edx user. I've made a new permission now to restrict it to only giving a response for THAT SAME user (as the one who's logged in and making the request), will push as soon as I rework the tests accordingly.

Reply:
Added now! We have a new permission that will restrict results to staff or to the user who is actually logged in (Optimizely will be sending the request on behalf of the user logged in). Thanks for the input- let me know if you have any questions!